driver-core/src/main/java/com/datastax/driver/core/QueryLogger.java

```diff
@@ -541,8 +541,9 @@ private int append(Statement statement, StringBuilder buffer, int remaining) {
             }
             remaining = append(" APPLY BATCH", buffer, remaining);
         } else {
-            // Unknown types of statement - should not happen
-            remaining = append(UNKNOWN_STATEMENT, buffer, remaining);
+            // Unknown types of statement
+            // Call toString() as a last resort
+            remaining = append(statement.toString(), buffer, remaining);
```

Comment:
Since we are introducing StatementWrapper, it will become more likely than I initially thought to find ourselves in the situation where the statement being logged is an unknown implementation. I thought it wiser to call toString() here instead of simply logging a dummy "??Unknown Statement??" string.